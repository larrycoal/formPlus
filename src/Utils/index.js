export const FormField = ({ input, handleChange }) => {
  let inputField = null;
  switch (input.type) {
    case "text":
      inputField = (
        <input
          type="text"
          value={input.value}
          name={input.name}
          placeholder={input.placeholder}
          onChange={(e) => {
            handleChange(e.target);
          }}
        />
      );
      break;
    case "select":
      inputField = (
        <span>
          <label>{input.name}</label>
          <select
            value={input.value}
            name={input.name}
            onChange={(e) => {
              handleChange(e.target);
            }}
          >
            {input.options.map((option, i) => (
              <option key={i} value={option}>
                {option}
              </option>
            ))}
          </select>
        </span>
      );
      break;
    default:
      return inputField;
  }
  return inputField;
};
export const TemplateCard = ({ template }, key) => (
  <div key={key}>
    <h2>{template.name}</h2>
    <p>{template.description}</p>
    <a href={template.link}>Use Template</a>
  </div>
);

export const filter = (state,filterInput)=>{   
    let newState = state
    if(filterInput.name==="category"){
         if(filterInput.value === "All"){
             return {
                filteredTemplate:newState,
                 category:filterInput.value
             }
         }else {
            let filtered = newState.filter((state)=>state.category === filterInput.value)
            return {
                filteredTemplate:filtered,
                category:filterInput.value,
                order:"Default",
                date:"Default"
            }
         }
    }
    else if(filterInput.name === "search"){
        let filtered = newState.filter((state)=>state.name.match(filterInput.value))
        return{
            filteredTemplate:filtered
        }
    }else if(filterInput.name === "order"){
        let def = newState
        let ordered = newState.sort((a,b)=>{
            let name1 = a.name.toUpperCase()
          let name2 = b.name.toUpperCase()
          if(name1 < name2){
            return -1
          }
          if(name1 > name2){
            return 1
          }
          return 0
        })
        if(filterInput.value === "Ascending"){
            return {
                filteredTemplate:ordered,
                order:filterInput.value
            }
        }
        else if(filterInput.value === "Descending"){
            return {
                filteredTemplate:ordered.reverse(),
                order:filterInput.value
            }
        }
        if(filterInput.value === "Default"){
            return {
                filteredTemplate:def,
                order:filterInput.value
            }
        }
    }
    else if(filterInput.name === "date"){
        let def = newState
        let ordered = newState.sort((a,b)=>{
            return a.date - b.date
        })
        if(filterInput.value === "Ascending"){
            console.log(ordered)
            return {
                filteredTemplate:ordered,
                date:filterInput.value
            }
        }
        else if(filterInput.value === "Descending"){
            return {
                filteredTemplate:ordered.reverse(),
                date:filterInput.value
            }
        }
        if(filterInput.value === "Default"){
            return {
                filteredTemplate:def,
                date:filterInput.value
            }
        }
    }
}
