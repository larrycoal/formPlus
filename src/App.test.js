import React from 'react'
import renderer from 'react-test-renderer'
import Form from './components/form'
import Pagination from './components/pagination'
import { TemplateCard } from './Utils'

const template = {
  name:"test",
  description:"test description",
  link:"https://test.com"
}
const key = "unique-key"

test("matches form snapshot",()=>{
  const tree = renderer.create(<Form></Form>).toJSON()
  expect(tree).toMatchSnapshot()
})
test("matches pagination snapshot",()=>{
  const tree = renderer.create(<Pagination></Pagination>).toJSON()
  expect(tree).toMatchSnapshot()
})
test("matches templateCard snapshot",()=>{
  const tree = renderer.create(<TemplateCard template={template} key={key}></TemplateCard>).toJSON()
  expect(tree).toMatchSnapshot()
})