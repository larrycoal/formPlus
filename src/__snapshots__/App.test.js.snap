// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches form snapshot 1`] = `
<div
  className="form_wrapper"
>
  <div>
    <input
      name="search"
      onChange={[Function]}
      placeholder="Search Templates"
      type="text"
      value=""
    />
    <i
      aria-hidden="true"
      class="fa fa-search"
    />
  </div>
  <div>
    <span>
      Sort By:
    </span>
    <span>
      <label>
        category
      </label>
      <select
        name="category"
        onChange={[Function]}
        value=""
      >
        <option
          value="All"
        >
          All
        </option>
        <option
          value="Health"
        >
          Health
        </option>
        <option
          value="E-commerce"
        >
          E-commerce
        </option>
        <option
          value="Education"
        >
          Education
        </option>
      </select>
    </span>
    <span>
      <label>
        order
      </label>
      <select
        name="order"
        onChange={[Function]}
        value=""
      >
        <option
          value="Default"
        >
          Default
        </option>
        <option
          value="Ascending"
        >
          Ascending
        </option>
        <option
          value="Descending"
        >
          Descending
        </option>
      </select>
    </span>
    <span>
      <label>
        date
      </label>
      <select
        name="date"
        onChange={[Function]}
        value=""
      >
        <option
          value="Default"
        >
          Default
        </option>
        <option
          value="Ascending"
        >
          Ascending
        </option>
        <option
          value="Descending"
        >
          Descending
        </option>
      </select>
    </span>
  </div>
</div>
`;

exports[`matches pagination snapshot 1`] = `
<div
  className="pagination_wrapper"
>
  <div>
    <button
      disabled={false}
      onClick={[Function]}
    >
      Previous
    </button>
  </div>
  <div>
    <span />
    of
    <span />
  </div>
  <div>
    <button
      disabled={false}
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;

exports[`matches template snapshot 1`] = `
<div>
  <h2>
    test
  </h2>
  <p>
    test description
  </p>
  <a
    href="https://test.com"
  >
    Use Template
  </a>
</div>
`;

exports[`matches templateCard snapshot 1`] = `
<div>
  <h2>
    test
  </h2>
  <p>
    test description
  </p>
  <a
    href="https://test.com"
  >
    Use Template
  </a>
</div>
`;
